### ISSUE TITLE
FP16 to FP8

### ISSUE THREAD

**Issue by usman9114** [ultralytics/yolov5#3043]
I figured on GPU model is already being quantized to FP16 by model.half() I was wondering how can I further quantize it to FP8. t.bfloat8() is not available. 

Thanks.

**Comment by glenn-jocher** [comment]
@usman9114 quantization depends on the hardware and the format of your model. The YOLOv5 iOS App for example uses FP8 quantized models.

**Comment by usman9114** [comment]
@glenn-jocher  I am running 9 instances of a custom trained model in parallel on Tesla V100. All inferences combined hitting only 10-20% of the GPU duty cycle before it hits the GPU memory limit of 16GB and max out. Currently size of custom model is about 170mb I was hoping if I could reduce it to 80mb by quantizing it to FP8.

**Comment by glenn-jocher** [comment]
@usman9114 sure, you can arbitrarily quantize also to any level, the design decision rests on the accuracy loss you're willing to accept for a given package size reduction.

**Comment by usman9114** [comment]
Thanks for your reply. I am not sure how can I quantize to FP8 so I was hoping if you have any pointers for me or a link to some tutorial will be of great help.

**Comment by glenn-jocher** [comment]
@usman9114 quantization will be constrained by your specific model format. You might want to visit the Export Tutorial and see models/export.py to get started.

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://docs.ultralytics.com/yolov5/tutorials/model_export)
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by usman9114** [comment]
I would like to quantize with default Pytorch format. Have you tried PyTorch 8 bit Post-Training Static Quantization ?? does it work with GPU ?

**Comment by glenn-jocher** [comment]
@usman9114 hmm, no, we use torch .half() or AMP quantization on GPU (CPU does not support non-FP32 torch ops), but have not tried 8 bits with torch. If you have success with this please let us know, thanks!

**Comment by PumeTu** [comment]
@usman9114 I am currently also trying to quantize yolo with the pytorch formats. Did you perhaps ever get it to work?
